### PR TITLE
Added <max_connections> to the websocket server

### PR DESCRIPTION
**Pull request by nkoenig**
This supports setting the maximum number of websocket connections.

Signed-off-by: Nate Koenig <nate@openrobotics.org>

### CODE REVIEW

File: plugins/websocket_server/WebsocketServer.cc [hunk 1]
```diff
@@ -214,6 +227,23 @@ bool WebsocketServer::Load(const tinyxml2::XMLElement *_elem)
...
+    try
+    {
+      this->maxConnections = std::stoi(elem->GetText());
+    }
+    catch (...)
+    {
+      ignerr << "Failed to convert port[" << elem->GetText() << "] to integer."
+        << std::endl;
+    }
+    igndbg << "Using maximum connetion count of "
```
Comment (german-e-mas) [inline plugins/websocket_server/WebsocketServer.cc:243]: Typo: connetion → connection

File: plugins/websocket_server/WebsocketServer.cc [hunk 2]
```diff
@@ -374,6 +404,9 @@ void WebsocketServer::OnConnect(int _socketId)
 //////////////////////////////////////////////////
 void WebsocketServer::OnDisconnect(int _socketId)
 {
+  if (this->connections.find(_socketId) == this->connections.end())
```
Comment (german-e-mas) [inline plugins/websocket_server/WebsocketServer.cc:407]: I would add the comment `Skip invalid sockets` here as well.

**Comment by nkoenig** [comment]
> LGTM! Left some minor comments.
> 
> So, if we set the max connections to 0, we basically prevent connections, right? That could come handy for the circuit runs I imagine.

Yeah, that's right.

File: plugins/websocket_server/WebsocketServer.cc (hunk 2 above)
Comment (nkoenig) [reply plugins/websocket_server/WebsocketServer.cc:408]: Fixed in a30524be7c154097aadcd62f4f840657efabf9f6

File: plugins/websocket_server/WebsocketServer.cc (hunk 1 above)
Comment (nkoenig) [reply plugins/websocket_server/WebsocketServer.cc]: Fixed in a30524be7c154097aadcd62f4f840657efabf9f6

**Pull request merged**